lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -3340,19 +3340,26 @@ swift::ASTContext *SwiftASTContext::GetASTContext() {
              moduleCachePath.c_str());
 
   // Compute the prebuilt module cache path to use:
-  // <resource-dir>/<platform>/prebuilt-modules
+  // <resource-dir>/<platform>/prebuilt-modules/<version>
   llvm::Triple triple(GetTriple());
-  llvm::SmallString<128> prebuiltModuleCachePath(GetResourceDir(triple));
-  StringRef platform;
-  if (swift::tripleIsMacCatalystEnvironment(triple)) {
-    // The prebuilt cache for macCatalyst is the same as the one for macOS,
-    // not iOS or a separate location of its own.
-    platform = "macosx";
-  } else {
-    platform = swift::getPlatformNameForTriple(triple);
-  }
-  llvm::sys::path::append(prebuiltModuleCachePath, platform,
-                          "prebuilt-modules");
+  llvm::Optional<llvm::VersionTuple> sdk_version =
+      m_ast_context_ap->LangOpts.SDKVersion;
+  if (!sdk_version) {
+    auto SDKInfoOrErr = clang::driver::parseDarwinSDKInfo(
+        *llvm::vfs::getRealFileSystem(),
+        m_ast_context_ap->SearchPathOpts.SDKPath);
+    if (SDKInfoOrErr) {
+      if (auto SDKInfo = *SDKInfoOrErr)
+        sdk_version = swift::getTargetSDKVersion(*SDKInfo, triple);
+    } else
+      llvm::consumeError(SDKInfoOrErr.takeError());
```

Comment:
should we log here? Is there any comment worth adding? As a reader, I am not clear what kind of errors this could be and whether this could be silencing a problem.

Reply:
The LLDB_LOG_ERROR  is perfect for this: 
```
    LLDB_LOG_ERROR(
        log, SDKInfoOrErr.takeError(),
        "Failed to get SDK info: {0}");
```

Reply:
The error would probably be that we are building for linux and the SDK doesn't have a version :-)
Since we are logging the SDK path itself, too, I'm hoping that the reason this fails would be very obvious.